File: app/src/main/java/repo/dao/NotificationDao.java
package repo.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import model.Notification;

@Dao
public interface NotificationDao {

    @Insert(entity = Notification.class, onConflict = OnConflictStrategy.REPLACE)
    void create(Notification notification);

    @Update(entity = Notification.class, onConflict = OnConflictStrategy.REPLACE)
    void update(Notification notification);

    @Delete(entity = Notification.class)
    void delete(Notification notification);


    @Query("SELECT * FROM notifications;")
    List<Notification> getAllNotifications();

    @Query("SELECT * FROM notifications WHERE is_done = 1;")
    List<Notification> getAllDoneNotifications();

    @Query("SELECT * FROM notifications WHERE id = :id;")
    Notification getNotification(int id);

    @Query("DELETE FROM notifications;")
    void deleteAllNotifications();

    @Query("DELETE FROM notifications WHERE is_done = 1;")
    void deleteAllDoneNotifications();

    @Query("SELECT COUNT(*) FROM notifications;")
    long getNotificationsCount();
}

File: app/src/main/java/repo/dao/NotificationDao.kt
package repo.dao

import androidx.room.*
import model.Notification

@Dao
interface NotificationDao {

    @Insert(entity = Notification::class, onConflict = OnConflictStrategy.REPLACE)
    suspend fun create(notification: Notification?)

    @Update(entity = Notification::class, onConflict = OnConflictStrategy.REPLACE)
    suspend fun update(notification: Notification?)

    @Delete(entity = Notification::class)
    suspend fun delete(notification: Notification?)

    @Query("SELECT * FROM notifications;")
    suspend fun getAllNotifications(): List<Notification>?

    @Query("SELECT * FROM notifications WHERE is_done = 1;")
    suspend fun getAllDoneNotifications(): List<Notification>?

    @Query("DELETE FROM notifications WHERE is_shown = 1;")
    suspend fun deleteShownNotifications()

    @Query("SELECT * FROM notifications WHERE id = :id;")
    suspend fun getNotification(id: Long): Notification?

    @Query("UPDATE notifications SET is_done = not is_done WHERE id = :id")
    suspend fun setDoneTodo(id: Long)

    @Query("UPDATE notifications SET is_shown = 1 WHERE id = :id;")
    suspend fun setShownTodo(id: Long)

    @Query("DELETE FROM notifications;")
    suspend fun deleteAllNotifications()

    @Query("DELETE FROM notifications WHERE is_done = 1;")
    suspend fun deleteAllDoneNotifications()

    @Query("SELECT * FROM notifications WHERE id = :id;")
    fun getNotificationThread(id: Long): Notification?

    @Query("SELECT * FROM notifications;")
    fun getAllNotificationsThread(): List<Notification>?
}